cli/tests/integrations/common.py

```diff
@@ -701,3 +701,41 @@ def base64_to_dict(byte_string):
     :rtype dict
     """
     return json.loads(base64.b64decode(byte_string).decode('utf-8'))
+
+
+UNIVERSE_REPO = "https://universe.mesosphere.com/repo"
+UNIVERSE_TEST_REPO = "http://universe.marathon.mesos:8085/repo"
+
+
+def setup_universe_server():
+    # add universe-server with static packages
+    assert_command(
+        ['dcos', 'marathon', 'app', 'add', 'tests/data/universe-v3-stub.json'])
+    watch_all_deployments()
+
+    assert_command(
+        ['dcos', 'package', 'repo', 'remove', 'Universe'])
+
+    assert_command(
+        ['dcos', 'package', 'repo', 'add', 'test-universe', UNIVERSE_TEST_REPO]
+    )
+
+    # Give the test universe some time to become available
+    describe_command = ['dcos', 'package', 'describe', 'helloworld']
+    for _ in range(10):
+        returncode, _, _ = exec_command(describe_command)
+        if returncode == 0:
+            break
+        time.sleep(1)
+    else:
+        teardown_universe_server()
```

Comment:
Nope! When the assertion is raised on the following line, PyTest does not call `teardown_module()` afterwards. This happened to me with local testing and I had to manually run the teardown commands. The library must assume that the setup code will not raise an exception.